nengo_gui/components/value.py

```diff
@@ -59,6 +59,9 @@ def update_client(self, client):
         # while there is data that should be sent to the client
         while len(self.data) > 0:
             item = self.data.popleft()
+            while len(self.data) > 0:
```

Comment:
Hmm, that's a lot less aggressive than the current code, in that it doesn't immediately send the data that has queued up while it's been busy putting together this list of data.  But I could do some testing to see if that less complete approach also works.